internal/bsr/summary.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package bsr
+
+import (
+	"context"
+	"fmt"
+)
+
+// SummaryAllocFunc is a function that returns a summary type
+type SummaryAllocFunc func(ctx context.Context) Summary
+
+type SummaryAllocFuncRegistry map[Protocol]map[ContainerType]SummaryAllocFunc
+
+func (r SummaryAllocFuncRegistry) Get(p Protocol, c ContainerType) (SummaryAllocFunc, bool) {
+	protocol, ok := r[p]
+	if !ok {
+		return nil, false
+	}
+	af, ok := protocol[c]
+	return af, ok
+}
+
+var SummaryAllocFuncs SummaryAllocFuncRegistry
```

Comment:
I use this in my unit test to test out the `Get` call
https://github.com/hashicorp/boundary/blob/f70aef0ee6b8fd6e2eae57287e2a8fb49b73e71c/internal/bsr/summary_test.go#L151

Reply:
We should refactor this test (or just define the new tests in a separate file), that imports `bsr` rather than `bsr_test` so it can call un-exported functions from the tests. Then this can be changed to be un-exported.

Reply:
Done